playbooks/aws/openshift-cluster/build_ami.yml

```diff
@@ -0,0 +1,83 @@
+---
+- hosts: localhost
+  connection: local
+  gather_facts: no
+  tasks:
+  - name: get the necessary vars for ami building
+    include_vars: vars.yml
+
+  - name: fetch the default subnet id
+    ec2_vpc_subnet_facts:
+      region: "{{ provision.region }}"
+      filters:
+        "tag:Name": "{{ provision.vpc.subnets[provision.region][0].az }}"
+    register: subnetout
+
+  - name: create instance for ami creation
+    ec2:
+      assign_public_ip: yes
+      region: "{{ provision.region }}"
+      key_name: "{{ provision.node_group_config.ssh_key_name }}"
+      #group_id: "{{ group_id }}"
+      group: "{{ provision.clusterid }}"
+      instance_type: m4.xlarge
+      vpc_subnet_id: "{{ subnetout.subnets[0].id }}"
+      image: "{{ provision.base_image }}"
+      volumes:
+      - device_name: /dev/sdb
+        volume_type: gp2
+        volume_size: 100
+        delete_on_termination: true
+      wait: yes
+      exact_count: 1
+      count_tag:
+        Name: ami_base
+      instance_tags:
+        Name: ami_base
+    register: amibase
+
+  - name: wait for ssh to become available
+    wait_for:
+      port: 22
+      host: "{{ amibase.tagged_instances.0.public_ip }}"
+      timeout: 300
+      search_regex: OpenSSH
+
+  - name: add host to group
+    add_host:
+      name: "{{ amibase.tagged_instances.0.public_dns_name }}"
+      groups: amibase
+
+- hosts: amibase
+  remote_user: root
```

Comment:
remote_user needs to be configurable if we support base rhel image to build from, probably should also add support for become_user in args somewhere.